lib/AST/LookupVisibleDecls.cpp

```diff
@@ -473,19 +473,70 @@ lookupVisibleMemberDeclsImpl(Type BaseTy, VisibleDeclConsumer &Consumer,
                              GenericSignatureBuilder *GSB,
                              VisitedSet &Visited);
 
-static void lookupVisibleProtocolMemberDecls(
-    Type BaseTy, ProtocolType *PT, VisibleDeclConsumer &Consumer,
-    const DeclContext *CurrDC, LookupState LS, DeclVisibilityKind Reason,
-                                             LazyResolver *TypeResolver, GenericSignatureBuilder *GSB,
-    VisitedSet &Visited) {
+class RestateFilteringConsumer : public VisibleDeclConsumer {
+  VisibleDeclConsumer &parentConsumer;
+  const Type baseTy;
+  const DeclContext *DC;
+  LazyResolver *resolver;
+  llvm::DenseSet<DeclName> foundVars;
+  llvm::DenseMap<std::tuple<DeclName, Type>, std::set<ValueDecl *>> foundFuncs;
+
+  void validate() {
+    assert(DC && baseTy && !baseTy->hasLValueType());
+  }
+public:
+  RestateFilteringConsumer(VisibleDeclConsumer &parentConsumer,
+                           Type baseTy, const DeclContext *DC,
+                           LazyResolver *resolver)
+  : parentConsumer(parentConsumer), baseTy(baseTy), DC(DC), resolver(resolver) {
+    validate();
+  }
+
+  void foundDecl(ValueDecl *VD, DeclVisibilityKind Reason) override {
+    assert(VD);
+    if (!VD->getDeclContext()->getAsProtocolOrProtocolExtensionContext()) {
```

Comment:
I don’t understand why this is specific to protocol extensions

Reply:
It isn't! I'll change this to `dyn_cast<ProtocolDecl>`